core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -380,14 +375,9 @@ private void commitReplaceTransaction(boolean orCreate) {
 
     } catch (RuntimeException e) {
       // the commit failed and no files were committed. clean up each update.
-      Tasks.foreach(updates)
-          .suppressFailureWhenFinished()
-          .run(
-              update -> {
-                if (update instanceof SnapshotProducer) {
-                  ((SnapshotProducer) update).cleanAll();
-                }
-              });
+      if (!ops.requireStrictCleanup() || e instanceof CleanableFailure) {
+        cleanUpOnCommitFailure();
```

Comment:
Shouldn't this be `cleanAllUpdates`? Both `cleanUpOnCommitFailure` and the `finally` block are going to run the delete function on every file.

Reply:
Oh yes good catch, we're duplicating the call to delete the files. Although I wonder how the create/replace tests for this case were passing.

Reply:
Ah the delete is best effort so the second time when the file no longer exists the delete would fail but we catch it and emit a warn log.